lib/components/navbar/NavLink.tsx

```diff
@@ -0,0 +1,12 @@
+import Link from 'next/link';
+import { withRouter } from 'next/router';
+
+const NavLink = (props: any) => (
```

Comment:
```suggestion
const NavLink : React.FC<NavLinkProps>= ({router, path, label}) => (
```
Or even better, reuse your already defined type:
```suggestion
const NavLink : React.FC<NavBarButton>= ({path, label}) => (
```
and then use `useRouter` in the body of the function.

Or, if you really want to use `withRouter` for some reason:
```suggestion
const NavLink : React.FC<NavBarButton & { NextRouter }>= ({router, path, label}) => (
```
Idk if its called `NextRouter` but you get the gist.